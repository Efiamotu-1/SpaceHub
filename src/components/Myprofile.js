import React from 'react';
import { useSelector } from 'react-redux';

const Myprofile = () => {
  const missions = useSelector((state) => state.stMissions);
  const rockets = useSelector((state) => state.stRockets.rockets);
  return (
    <div>
      <div className="mt-5 flex mx-auto w-11/12 space-x-5">
        {/* Missions Filter & render */}
        <div className="w-1/2">
          <p className="font-thin text-3xl"> My Rockets </p>
          <div className="border mt-3 rounded">
            {missions.filter((mission) => mission.reserved === true).map((mission) => (mission && <div className="px-4 py-2 border">{mission.mission_name}</div>))}

          </div>
        </div>

        {/* Rockets Filter & render */}
        <div className="w-1/2">
          <p className="font-thin text-3xl">My Rockets</p>
          <div className="border mt-3 rounded">
            {rockets.filter((rocket) => rocket.reserved === true).map((rocket) => (rocket && <div className="px-4 py-2 border">{rocket.name}</div>))}

          </div>
        </div>
      </div>
    </div>
  );
};

export default Myprofile;
