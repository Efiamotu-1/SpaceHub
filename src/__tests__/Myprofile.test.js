import React from 'react';
import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import store from '../redux/configStore';
import Myprofile from '../components/Myprofile';
import '@testing-library/jest-dom';

describe('Myprofile Component', () => {
  test('Correct render of Myprofile', () => {
    const myprofile = render(
      <React.StrictMode>
        <Provider store={store}>
          <Router>
            <Myprofile />
          </Router>
        </Provider>
      </React.StrictMode>,
    );
    expect(myprofile).toMatchSnapshot();
  });

  test('Check if \'My Missions\' is in the rendered document of Myprofile component', () => {
    render(
      <React.StrictMode>
        <Provider store={store}>
          <Router>
            <Myprofile />
          </Router>
        </Provider>
      </React.StrictMode>,
    );
    expect(screen.getByText('My Missions')).toBeInTheDocument();
  });

  test('Check if \'My Rockets\' is in the rendered document of Myprofile component', () => {
    render(
      <React.StrictMode>
        <Provider store={store}>
          <Router>
            <Myprofile />
          </Router>
        </Provider>
      </React.StrictMode>,
    );
    expect(screen.getByText('My Rockets')).toBeInTheDocument();
  });
});
