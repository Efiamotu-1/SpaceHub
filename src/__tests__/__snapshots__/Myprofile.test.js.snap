// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Myprofile Component Correct render of Myprofile 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="mt-5 flex mx-auto w-11/12 space-x-5"
        >
          <div
            class="w-1/2"
          >
            <p
              class="font-thin text-3xl"
            >
              My Missions
            </p>
            <div
              class="border mt-3 rounded"
            />
          </div>
          <div
            class="w-1/2"
          >
            <p
              class="font-thin text-3xl"
            >
              My Rockets
            </p>
            <div
              class="border mt-3 rounded"
            />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="mt-5 flex mx-auto w-11/12 space-x-5"
      >
        <div
          class="w-1/2"
        >
          <p
            class="font-thin text-3xl"
          >
            My Missions
          </p>
          <div
            class="border mt-3 rounded"
          />
        </div>
        <div
          class="w-1/2"
        >
          <p
            class="font-thin text-3xl"
          >
            My Rockets
          </p>
          <div
            class="border mt-3 rounded"
          />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
